### DISCUSSION TITLE
[Bug]: Migrating pimcore v 6.9.6 to pimcore 10.6.3 v, facing issue

### DISCUSSION THREAD

**Discussion by Thejaswini-Rao-U** [pimcore/pimcore#15620] [Q&A / Support]
### Pimcore version

Pimcore migrating to 10.6.3 v

### Steps to reproduce

Hi 

Im updating pimcore v 6.9.6 to 10.6.3v , 

Initially did all the changes to the folder structures manually and executed this command COMPOSER_MEMORY_LIMIT=-1 composer update

### Actual Behavior

while migrating im getting below error which is attached
im using php 8.1 v and mysql 8.0v
composer version is 2.5v

**[error] Migration Pimcore\Bundle\CoreBundle\Migrations\Version20210107103923 failed during Execution. Error: "Please provide select options as associative array with fields "key" and "value""

In Select.php line 206:**
  Please provide select options as associative array with fields "key" and "value"**                                         
![keyValue](https://github.com/pimcore/pimcore/assets/78895325/a6575453-7a2a-4e9b-9b44-79eb456626e2)

Please help me to solve this issue
Thanks in advance

### Expected Behavior

Successfully it should migrate

**Comment by NiklasBr** [comment]
It looks like perhaps one of your data oject classes has a select/multiselect field whose values are not in the expected format.

Check the `pimcore/var/classes/definition_Classwitherror.php` it should have some `options` looking something like this 

```php
'options' => 
    array (
        0 => 
                  array (
                    'key' => 'Closing down',
                    'value' => 'closing',
                  ),
        1 => 
                  array (
                    'key' => 'Opening',
                    'value' => 'opening',
                  ),
```

**Comment by Thejaswini-Rao-U** [comment]
@NiklasBr , Thanks for the quick resonse.

i have checked all the classes options provider , all have similar expected type of array. 


array:81 [
  0 => array:2 [
    "key" => "NAVY LIGHT BLUE"
    "value" => "NLB"
  ]
  1 => array:2 [
    "key" => "OLIVE"
    "value" => "OL"
]
]

But Im not sure why this error still exists?

**Comment by NiklasBr** [comment]
Are you able to add a debugger breakpoint where the error happens? 

https://github.com/pimcore/pimcore/blob/947054028b6fbfe58b4ced996f88fc7614ab5cb4/models/DataObject/ClassDefinition/Data/Select.php#L116-L133

Looks like maybe `WebsiteCategory` or an optionsprovider is the culprit.

**Comment by Thejaswini-Rao-U** [comment]
@NiklasBr , Even if i remove all the class option provider and classes definition, still i get the issue.

is it due to PHP deprecation error? earlier version we have used 7.2 and now switched to 8.1 when migrating

![keyValue](https://github.com/pimcore/pimcore/assets/78895325/3a76bfee-d7d4-4f00-ba11-31467489ad3e)

**Comment by NiklasBr** [comment]
I can't see any reason why the PHP version would have an effect. Are you able to provide the class definition which fails here?

**Comment by dvesh3** [comment]
Moved it discussions as the issue seems specific to your application data. I would suggest to execute the failing migration with verbose and check where it fails. thanks!

**Reply by Thejaswini-Rao-U** [reply]
can you explain breifly?

**Comment by Thejaswini-Rao-U** [comment]
@NiklasBr , i cannot see any errors in specific classes, even if i rename the class definition and check, still the error comes. I think need to go deeper into this issue, have to debug all the data there

**Comment by Thejaswini-Rao-U** [comment]
Hi
 Anyone can tell me what is the wrong here ?
  There is no extension able to load the configuration for "_pimcore_dev" (in "/var/www/html/amrod-config/packages/dev/routes.yaml"). Looked for namespace "_pimcore_dev", found ""framework", "security", "twig", "twig_extra", "monolog", "doctrine", "doctrine_migratio  
  ns", "sensio_framework_extra", "cmf_routing", "presta_sitemap", "scheb_two_factor", "fos_js_routing", "flysystem", "debug", "web_profiler  
  ", "pimcore", "pimcore_admin", "elements_process_manager"" in /var/www/html/amrod-upgrade10/config/packages/dev/routes.yaml (which is bei  
  ng imported from "/var/www/html/amrod-upgrade10/src/Kernel.php"). 
  
  routes.yaml fine has the content below
  
  _pimcore_dev:
    resource: "@PimcoreCoreBundle/Resources/config/routing_dev.yaml"

_main:
    resource:  ../../routes.yaml

**Comment by Thejaswini-Rao-U** [comment]
Hi Anyone Know why this error arries?
what wrong i did here?

![error](https://github.com/pimcore/pimcore/assets/78895325/f5c4155e-a34f-403c-9f89-b7bd7b286472)

**Reply by Thejaswini-Rao-U** [reply]
@NiklasBr , any idea ?

**Comment by Thejaswini-Rao-U** [comment]
Anyone please help me to organize existing bundle which was there inside src/bundleName1/, src/bundleName2/ path, where to move it ?

i had moved already src/AppBundle/ to src/ path, other bundles code can i make separate bundle folder and move it outside src/ path?

Please suggest? im stuck in this migration

**Comment by Thejaswini-Rao-U** [comment]
Hi All
I have solved above issue everything. But now the problem is when i hit the admin url, it gives me below error
![router](https://github.com/pimcore/pimcore/assets/78895325/8271e609-3985-4ff9-89a9-9b019dbd89e2)


anyone come across this error? can anyone suggest the solution ? i have installed symfony profiler

**Reply by dvesh3** [reply]
please make sure you have `routes.yaml` in root config folder 
![image](https://github.com/pimcore/pimcore/assets/5137917/2e2c08e2-6bbc-4e3d-9e23-3946b76e3abe)

which requires core bundle `routing_dev.yaml`
![image](https://github.com/pimcore/pimcore/assets/5137917/5c32c20e-1aff-49ed-90d8-b6bbd68848dc)

I would recommend you to cross check your project structure with demo app since you are facing so many issues https://github.com/pimcore/demo/tree/10.3/config

**Comment by Thejaswini-Rao-U** [comment]
@dvesh3 , Thanks for the response
As you mentioned all the files i have it in directory structure, initially i have organized folder structure according to the pimcore 10.6 document and i have mapped with the skeleton github projecet. after this, i have migrated successfully.

Issue is not with the structure.

Its with the profiler, i have installed and its working.. redirecting into login page,

now the issue is im not able to login with admin, even after resetting the admin password.. Its redirecting into login page itself.

**Comment by Thejaswini-Rao-U** [comment]
Its fixed
